### PR TITLE
✨ Add avatar from url when register

### PR DESCRIPTION
**Pull request by williamchong**
tested on dev servers, should be working no problem

### CODE REVIEW

File: src/util/fileupload.js [hunk 1]
```diff
@@ -54,4 +55,27 @@ export async function handleAvatarUploadAndGetURL(user, file, avatarSHA256) {
...
+export async function handleAvatarLinkAndGetURL(user, url) {
+  const { data } = await axios.get(url, {
+    responseType: 'arraybuffer',
+    timeout: 5000,
+  });
+  const buffer = new Uint8Array(data);
+  const type = fileType(buffer);
+  if (!SUPPORTED_AVATER_TYPE.has(type && type.ext)) {
+    console.error(`unsupported file format! ${(type || {}).ext || JSON.stringify(type)}`);
+    return undefined;
```
Comment (nnkken) [inline src/util/fileupload.js]: Should it throw error and let caller handle it?

File: src/util/fileupload.js [hunk 2]
```diff
@@ -54,4 +55,27 @@ export async function handleAvatarUploadAndGetURL(user, file, avatarSHA256) {
...
+  }
+  const resizedBuffer = await sharp(data).resize(400, 400).toBuffer();
+  const filename = url.split('/').pop();
+  const file = {
+    buffer: resizedBuffer,
+    filename,
+    originalname: filename,
```
Comment (nnkken) [inline src/util/fileupload.js]: seems unnecessary since you defined newFilename for `uploadFileAndGetLink`?

File: src/util/fileupload.js [hunk 3]
```diff
@@ -54,4 +55,27 @@ export async function handleAvatarUploadAndGetURL(user, file, avatarSHA256) {
...
 }
 
+export async function handleAvatarLinkAndGetURL(user, url) {
+  const { data } = await axios.get(url, {
+    responseType: 'arraybuffer',
+    timeout: 5000,
+  });
+  const buffer = new Uint8Array(data);
+  const type = fileType(buffer);
+  if (!SUPPORTED_AVATER_TYPE.has(type && type.ext)) {
```
Comment (nwingt) [inline src/util/fileupload.js]: `AVATER`

File: src/util/fileupload.js (hunk 1 above)
Comment (williamchong) [reply src/util/fileupload.js]: I think this was done to mitigate invalid avatar blocking user registration a while ago

**Review by nnkken — APPROVED** [review]

**Pull request merged**